src/opera/util/input_validation.py

```diff
@@ -130,6 +131,133 @@ def validate_slc_s1_inputs(runconfig, logger, name):
             logger.critical(name, ErrorCode.INVALID_INPUT, error_msg)
 
 
+def get_burst_id_set(input_file_group, logger, name):
+    """
+    Compiles a set of burst_ids from a list of files defined in the runconfig file.
+    Each file in the list should have have a burst_id in the file name.
+
+    Parameters
+    ----------
+    input_file_group : list
+        List of files containing a burst_id
+    logger : PgeLogger
+        Logger passed by the calling PGE
+    name : str
+        PGE name
+
+    Returns
+    -------
+    burst_ids :set
+
+    Raises
+    ------
+    RuntimeError
+        If a burst_id is improperly formatted.
+
+    """
+    burst_ids = set()
+    for i in input_file_group:
+        reg_ex = r't\w{3}_\d{6}_iw[1|2|3]'
+        match = re.search(reg_ex, i)
+        if match:
+            burst_ids.add(re.findall(reg_ex, i)[0])
+        else:
+            msg = f'Input file present without properly formatted burst_id: {i}'
+            logger.critical(name, ErrorCode.INVALID_INPUT, msg)
+
+    return burst_ids
+
+
+def check_amp_mean_geometry_burst_ids(cslc_input_burst_ids, ancillary_file_list, logger, name):
+    """
+    Verify burst_ids from the ancillary input files:
+        'amplitude_dispersion_files',
+        'amplitude_mean_files',
+        'geometry_files
+    Verify that each file has a unique burst id in the file name, and that the burst idea is in the
+    set of of DISP_S1 cslc input burst ids.
+
+    Parameters
+    ----------
+    cslc_input_burst_ids : set
+        Set of the DISP_S1 input file burst_ids
+    ancillary_file_list : list
+        List from the runconfig file of the desired set of ancillary files to check
+    logger : PgeLogger
+        Logger passed by the calling PGE
+    name : str
+        PGE name
+
+    Raises
+    ------
+    RuntimeError
+        If there are ancillary files of the same type with the same burst_id.
+        If the set of ancillary file burst_ids do not match the set of cslc input file burst_ids.
+
+    """
+    ancillary_burst_ids = get_burst_id_set(ancillary_file_list, logger, name)
+    # Test that there is a unique burst_id for each burst
+    if len(ancillary_burst_ids) != len(ancillary_file_list):
+        msg = f'Burst ids in {ancillary_file_list} are not unique.'
+        logger.critical(name, ErrorCode.INVALID_INPUT, msg)
+    # Verify that the sets of burst_ids are equal
+    if ancillary_burst_ids != cslc_input_burst_ids:
+        msg = f'input_burst_ids: {cslc_input_burst_ids} do not match ancillary_burst_ids: ' \
```

Comment:
The formatting of this error message might be hard to read in production where there could be up to 30 burst ids printed out from each set, so we might want to add some line breaks and only print out the differences between the two sets, like so:

```python
msg = ("Input CSLC burst IDs do not match the set of ancillary burst IDs\n"
        f"{cslc_input_burst_ids.difference(ancillary_burst_ids)=\n}
        f"{ancillary_burst_ids.difference(cslc_input_burst_ids)=\n}")
```